dali/kernels/imgproc/resample/resampling_batch.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_RESAMPLING_BATCH_H_
+#define DALI_KERNELS_IMGPROC_RESAMPLING_BATCH_H_
+
+#include "dali/kernels/imgproc/resample/resampling_setup.h"
+
+namespace dali {
+namespace kernels {
+
+template <int which_pass, typename Output, typename Input>
```

Comment:
This function is only declared here and is explicitly instantiated for common cases.
In general, it is assumed that the intermediate representation is always `float`, so the available instantiations are always `<which_pass = 0, Output = float, Input>` or `<which_pass = 1, Output, Input = float>`.

Reply:
From GCG: 
> Template parameters should follow the naming style for their category: type template parameters should follow the rules for type names, and non-type template parameters should follow the rules for variable names.